examples/third_party/openssl/BUILD.openssl.bazel

```diff
@@ -15,37 +15,74 @@ CONFIGURE_OPTIONS = [
     "no-comp",
     "no-idea",
     "no-weak-ssl-ciphers",
+    "no-shared",
 ]
 
-configure_make(
+LIB_NAME = "openssl"
+
+MAKE_TARGETS = [
+    "build_libs",
+    "install_dev",
+]
+
+alias(
     name = "openssl",
+    actual = select({
+        "@rules_foreign_cc_examples//:msvc_compiler": "openssl_msvc",
+        "//conditions:default": "openssl_default",
+    }),
+    visibility = ["//visibility:public"],
+)
+
+configure_make_variant(
+    name = "openssl_msvc",
+    build_data = [
+        "@nasm//:nasm",
+        "@perl//:perl",
+    ],
+    configure_command = "Configure",
+    configure_in_place = True,
+    configure_options = CONFIGURE_OPTIONS + [
+        "VC-WIN64A",
+        # Unset Microsoft Assembler (MASM) flags set by built-in MSVC toolchain,
+        # as NASM is unsed to build OpenSSL rather than MASM
+        "ASFLAGS=\" \"",
+    ],
+    configure_prefix = "$PERL",
+    env = {
+        # The Zi flag must be set otherwise OpenSSL fails to build due to missing .pdb files
+        "CFLAGS": "-Zi",
```

Comment:
Is there a way to avoid overwriting `CFLAGS` here? I think it'd be nice to allow Bazel to continue to define them.

Reply:
The `configure_make` rule will append `-Zi` to the `CFLAGS`, rather than overwrite them :)